inference-engine/tests/functional/plugin/shared/src/single_layer_tests/eltwise.cpp

```diff
@@ -7,149 +7,130 @@
+    std::ostringstream results;
+
+    results << "IS=" << CommonTestUtils::vec2str(inputShapes) << "_";
+    results << "eltwiseOpType=" << EltwiseParams::EltwiseOpType_to_string(eltwiseOpType) << "_";
+    results << "secondaryInputType=" << EltwiseParams::InputLayerType_to_string(secondaryInputType) << "_";
+    results << "opType=" << EltwiseParams::OpType_to_string(opType) << "_";
+    results << "netPRC=" << netPrecision.name() << "_";
+    results << "targetDevice=" << targetName;
+    return results.str();
 }
 
 void EltwiseLayerTest::SetUp() {
-    EltwiseOpType op;
-    ParameterInputIdx primary_input_idx;
-    InputLayerType secondary_input_type;
-    InferenceEngine::SizeVector inputShape;
+    std::vector<std::vector<size_t>> inputShapes;
     InferenceEngine::Precision netPrecision;
-    ngraph::ParameterVector parameter_inputs;
+    EltwiseParams::InputLayerType secondaryInputType;
+    EltwiseParams::OpType opType;
+    EltwiseParams::EltwiseOpType eltwiseOpType;
     std::map<std::string, std::string> additional_config;
-    std::tie(op, primary_input_idx, secondary_input_type, netPrecision, inputShape, targetDevice, additional_config) = this->GetParam();
-    configuration.insert(additional_config.begin(), additional_config.end());
+    std::tie(inputShapes, eltwiseOpType, secondaryInputType, opType, netPrecision, targetDevice, additional_config) = this->GetParam();
     auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
 
-    std::shared_ptr<ngraph::Node> input0_node;
-    std::shared_ptr<ngraph::Node> input1_node;
-    auto primary_input = ngraph::builder::makeParams(ngPrc, { inputShape })[0];
-
-    switch (secondary_input_type) {
-    case InputLayerType::CONSTANT:
-    {
-        auto shape_total = 1;
-        for (auto dim : inputShape) {
-            shape_total *= dim;
-        }
+    configuration.insert(additional_config.begin(), additional_config.end());
+    auto input = ngraph::builder::makeParams(ngPrc, {inputShapes[0]});
+
+    std::vector<size_t> shape_input_secondary;
+    switch (opType) {
+        case EltwiseParams::OpType::SCALAR:
+            shape_input_secondary = std::vector<size_t>({1});
+            break;
+        case EltwiseParams::OpType::VECTOR:
+            shape_input_secondary = inputShapes[0];
+            break;
+        default:
+            FAIL() << "Unsupported SubtractionType: " << EltwiseParams::OpType_to_string(opType);
+    }
 
-        const float min = -10;
-        const float max = 10;
-        const float range = max - min;
-        const float step = range / shape_total;
+    std::shared_ptr<ngraph::Node> secondary_input;
+    switch (secondaryInputType) {
+        case EltwiseParams::InputLayerType::CONSTANT:
+            secondary_input = ngraph::builder::makeConstant(ngPrc, shape_input_secondary, std::vector<float>{-1.0f});
```

Comment:
I've briefly overviewed our existing data generators since @kmagiers requested above, to generate constants with real floats but not with float-like ints,  and found that all off [genrateVector functions](https://github.com/openvinotoolkit/openvino/blob/cfa87999e390781bccfe84987f3cdefb55928ccf/inference-engine/tests/ngraph_functions/include/ngraph_functions/utils/data_utils.hpp) generates exactly float-like-ints vectors, so I propose to use e.g. https://github.com/openvinotoolkit/openvino/blob/4f6c976add457e3013d25fd15ada49f56a6d0521/inference-engine/tests/ie_test_utils/common_test_utils/data_utils.hpp#L27